src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts

```diff
@@ -496,12 +497,29 @@ export class HostsComponent extends ListWithDetails implements OnDestroy, OnInit
           hostList.map((host) => {
             const counts = {};
             host['service_instances'] = new Set<string>();
-            host['services'].forEach((service: any) => {
-              counts[service.type] = (counts[service.type] || 0) + 1;
-            });
-            host['services'].map((service: any) => {
-              host['service_instances'].add(`${service.type}: ${counts[service.type]}`);
-            });
+            if (this.orchStatus?.available) {
+              let daemons: Daemon[] = [];
+              let observable: Observable<Daemon[]>;
+              observable = this.hostService.getDaemons(host['hostname']);
+              observable.subscribe((dmns: Daemon[]) => {
+                daemons = dmns;
+                daemons.forEach((daemon: any) => {
+                  counts[daemon.daemon_type] = (counts[daemon.daemon_type] || 0) + 1;
+                });
+                daemons.map((daemon: any) => {
+                  host['service_instances'].add(
+                    `${daemon.daemon_type}: ${counts[daemon.daemon_type]}`
+                  );
+                });
+              });
+            } else {
+              host['services'].forEach((service: any) => {
+                counts[service.type] = (counts[service.type] || 0) + 1;
+              });
+              host['services'].map((service: any) => {
+                host['service_instances'].add(`${service.type}: ${counts[service.type]}`);
+              });
+            }
```

Comment:
can't we do this process in backend? @avanthakkar

Reply:
@avanthakkar i think we can afford a refactor for this now right? I mean we've been (including myself) have been patching up this area for a while and I think this deserves some cleaning up. Would you mind looking into the cleanup again and give it a try?